packages/ragbits-document-search/src/ragbits/document_search/retrieval/rerankers/litellm.py

```diff
@@ -0,0 +1,51 @@
+from typing import List
+
+import litellm
+
+from ragbits.document_search.documents.element import Element, TextElement
+from ragbits.document_search.retrieval.rerankers.base import Reranker
+
+
+class LiteLLMReranker(Reranker):
+    """
+    A LiteLLM reranker for providers such as Cohere, Together AI, Azure AI.
+    """
+
+    model: str
+    top_n: int | None = None
+    return_documents: bool = False
+    rank_fields: list[str] | None = None
+    max_chunks_per_doc: int | None = None
+
+    def rerank(self, chunks: List[Element], query: str) -> List[Element]:
+        """
+        Reranking with LiteLLM API.
+
+        Args:
+            chunks: The chunks to rerank.
+            query: The query to rerank the chunks against.
+
+        Returns:
+            The reranked chunks.
+
+        Raises:
+            ValueError: If chunks are not a list of TextElement objects.
+        """
+        if not all(isinstance(chunk, TextElement) for chunk in chunks):
+            raise ValueError("All chunks must be TextElement instances")
+
+        documents = [chunk.content if isinstance(chunk, TextElement) else None for chunk in chunks]
+
+        response = litellm.rerank(
```

Comment:
maybe it would be better to use `arerank` here: https://docs.litellm.ai/docs/rerank#async-usage, what do you think?